uint128/uint128.go

```diff
@@ -0,0 +1,121 @@
+package uint128
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type Int [2]uint64
+
+func SetUint64(lo uint64) *Int {
+	// i[1] = high bits
+	// i[0] = low bits
+	z := &Int{}
+	z.setUint128(0, lo)
+	return z
+}
+func SetUint128(hi, lo uint64) *Int {
+	z := &Int{}
+	z.setUint128(hi, lo)
+	return z
+}
+
+func SetInt128(hex_str string) *Int {
```

Comment:
Let's do what felt does:

1. Implement two functions: `SetBigInt` and `SetString`
2. In `SetString`, convert the string to a big int and then use `SetBigInt` to convert to `uint128.Int`. This way we automatically support all the string formats that `math/big.Int.SetString` supports